.travis.yml

```diff
@@ -3,24 +3,24 @@ dist: xenial
 matrix:
     include:
         - python: 2.7
+          env: TOXENV=py27
         - python: 3.5
+          env: TOXENV=py35
         - python: 3.6
+          env: TOXENV=py36
         - python: 3.7
+          env: TOXENV=py37
         - python: 3.8-dev
+          env: TOXENV=py38
         - name: "Linting"
           python: 3.7
-          env: TARGET=lint-ci
+          env: TOXENV=lint
           cache:
               directories:
                   - $HOME/.cache/pre-commit
     allow_failures:
         - python: 3.8-dev
-before_install: pip install -U codecov
-install:
-    - if [[ $TRAVIS_PYTHON_VERSION == '2.7' ]]; then make build27; fi
-    - if [[ $TRAVIS_PYTHON_VERSION == '3.5' ]]; then make build35; fi
-    - if [[ $TRAVIS_PYTHON_VERSION == '3.6' ]]; then make build36; fi
-    - if [[ $TRAVIS_PYTHON_VERSION == '3.7' ]]; then make build37; fi
-    - if [[ $TRAVIS_PYTHON_VERSION == '3.8-dev' ]]; then make build38; fi
-script: make "${TARGET:-test}"
+          env: TOXENV=py38
+install: pip install -U codecov tox
+script: tox --recreate
```

Comment:
I don't think you need `--recreate` here, I'm actually not sure why pytest does this 🤔

Reply:
I was wondering that as well. I thought it was unnecessary since each Travis run executes in a fresh VM.

Reply:
Also, what do you think about adding the `TOXENV=py38` to `allow_failures` and `Linting`? I was unsure if I needed it in both.

Reply:
looks like it's been there since the beginning of time in the pytest repo -- you don't need it (I'll go remove it from there too!)

Reply:
your `allow_failures` looks correct to me :+1:

Reply:
Oh sorry, I meant repeating `TOXENV=py38` in both `allow_failures` and in the `3.8-dev` target. They are both testing the same environment, so I was wondering if I need to repeat it. Let me try.

Reply:
yeah you need it in both

Reply:
Interesting. I wonder why hah

Reply:
travis is picky and requires the whole mapping to match